app/charts/shared/chart-state.ts

```diff
@@ -53,13 +55,13 @@ export const getChartStateMetadata = (
           );
       }
     case "line":
-      return getLinesStateMetadata(chartConfig);
+      return getLinesStateMetadata(chartConfig, observations, dimensions);
     case "map":
       return getMapStateMetadata();
     case "pie":
-      return getPieStateMetadata();
+      return getPieStateMetadata(chartConfig, observations, dimensions);
     case "scatterplot":
-      return getScatterplotStateMetadata();
+      return getScatterplotStateMetadata(chartConfig, observations, dimensions);
```

Comment:
So I guess that if we have the same getMetadata method everywhere, we can remove the switch.